framework/lib/components/createable-select-dropdown/createable-select-dropdown.tsx

```diff
@@ -125,29 +135,31 @@ export const CreatableSelectDropdown = <T extends string = string> ({
     },
   })
 
-  const handleInputChange = (newValue: string) => {
-    setNewOptionText(newValue)
-  }
-
-  const handleCreateOption = () => {
-    const newOptionTextLower = newOptionText.toLowerCase()
-    const newOptionExists = standardOptions.some((option) =>
-      option.value.toLowerCase() === newOptionTextLower)
+  const handleCreateOption = (newOptionTextString: string = '') => {
+    const newOptionTextLower = newOptionTextString.toLowerCase()
+    const newOptionExists = standardOptions.some(
+      (option) => option.value.toLowerCase() === newOptionTextLower
+    )
 
     if (!newOptionExists) {
-      const newOption = { label: newOptionText, value: newOptionText }
-      setCreatedOptions((currentCreatedOptions) => [ ...currentCreatedOptions, newOption ])
+      const newOption = {
+        label: newOptionTextString,
+        value: newOptionTextString,
+      }
+      setCreatedOptions((currentCreatedOptions) => [
+        ...currentCreatedOptions,
+        newOption,
+      ])
       setSelectedOption(newOption)
       onOptionChange(newOption)
     }
-
-    setNewOptionText('')
     setNewOptionPlaceholder(initialPlaceholder)
   }
 
   const handleChange = (
     newValue: SingleValue<Option>,
-    _actionMeta: ActionMeta<Option>) => {
+    _actionMeta: ActionMeta<Option>
+  ) => {
     const option = newValue
 
     if (option === null) {
```

Comment:
Also, one thing that has been there before but would need fixing. If you try selecting add field by pressing space or enter when focused on that option, it will actually select 'add field'. Check in the handleChange if `newValue` is equal to Add field, and toggleCreateOption then instead